File: benchmark/comparison/results/adv_ts/09.ts
function process_data(lst: number[]): number {
  return lst
    .filter(n => n >= 0)
    .map(n => n * 2)
    .reduce((a, b) => a + b, 0);
}

function run(lst: number[]): void {
  console.log(`Total: ${process_data(lst)}`);
}

run([1, 2, 3]);
run([-1, -2]);
run([5, -3, 5]);

File: benchmark/comparison/results/adv_py/09.py
def process_data(lst):
    filtered = [x for x in lst if x >= 0]
    doubled = [x * 2 for x in filtered]
    return sum(doubled)

def run(lst):
    print(f"Total: {process_data(lst)}")

run([1, 2, 3])
run([-1, -2])
run([5, -3, 5])
